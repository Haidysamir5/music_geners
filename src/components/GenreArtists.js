import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchArtists } from "../actions";
import ArtistDetials from "./ArtistDetails";

class GenreArtists extends Component {
    update=false;
  componentDidMount() {
    const { id } = this.props;
    this.props.fetchArtists(id);
  }

componentWillUpdate(){
    this.update=true;
}
  renderArtists() {
    var { artists } = this.props;
    // pathname
    // window.onhashchange
    if (artists.length > 0 && this.update) {
      return artists.map((artist) => {
        return <ArtistDetials artist={artist} key={artist.id} />;
      });
    }
    return "...loading";
  }

  render() {
    return (
      <div className="container">
        <div className="row">{this.renderArtists()}</div>
      </div>
    );
  }
}
function mapStateToProps(state) {
  return { artists: state.artists };
}

export default connect(mapStateToProps, { fetchArtists })(GenreArtists);
