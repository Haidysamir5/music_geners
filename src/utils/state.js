export const music={
    modal_state:false,
    music_info:[]
}