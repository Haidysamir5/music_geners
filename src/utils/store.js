import { createStore, applyMiddleware } from "redux";
import promiseMiddleware from "redux-promise";
import reducers from "../reducers";
export const store = createStore(reducers, applyMiddleware(promiseMiddleware));

