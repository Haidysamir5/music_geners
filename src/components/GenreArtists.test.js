import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";
import { act } from "react-dom/test-utils";
import GenreArtists from "./GenreArtists";
// wrap the root component in a Provider
import { store } from "../utils/store";

const fetchArtists = jest.fn();
jest.mock("../actions", async () => {
  return {
    __esModule: true,
    default: async () =>  { props:[
      
      {
        id: 0,
        name: "All",
        picture_big: "https.jpg",
        type: "genre",
      },
      {
        id: 132,
        name: "Pop",
        picture_big: "https:.jpg",

        type: "genre",
      }
    ]}
  };
});

test("match snapshoot", async () => {
  let wrapper;
  await act(async () => {
    wrapper = mount(<GenreArtists store={store} fetchArtists={fetchArtists}/>);
  });
  wrapper.update();

  expect(toJson(wrapper)).toMatchSnapshot();
});
