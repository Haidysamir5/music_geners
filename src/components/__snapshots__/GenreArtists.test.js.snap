// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshoot 1`] = `
<Connect(GenreArtists)
  fetchArtists={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <GenreArtists
    artists={Array []}
    fetchArtists={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        ...loading
      </div>
    </div>
  </GenreArtists>
</Connect(GenreArtists)>
`;
